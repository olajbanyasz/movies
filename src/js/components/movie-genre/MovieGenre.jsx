import React, { Component } from 'react';
import './moviegenre.style.less';

class MovieGenre extends Component {
  render() {
    return (
      <div className='movie-genre'>
        {this.props.genres}
      </div>
    );
  }
}

export default MovieGenre;