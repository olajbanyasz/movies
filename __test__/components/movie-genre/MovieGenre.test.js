import React from 'react';
import { shallow } from 'enzyme';
import MovieGenre from '../../../src/js/components/movie-genre/MovieGenre';

describe('MovieGenre', () => {
  it('should rendered correctly', () => {
    const component = shallow(<MovieGenre genres={['Drama', 'Sci-Fi']} />);
    expect(component).toMatchSnapshot();
  });
});
