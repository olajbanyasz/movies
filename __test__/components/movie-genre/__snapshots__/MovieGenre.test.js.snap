// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieGenre should rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieGenre
    genres={
      Array [
        "Drama",
        "Sci-Fi",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Drama",
        "Sci-Fi",
      ],
      "className": "movie-genre",
    },
    "ref": null,
    "rendered": Array [
      "Drama",
      "Sci-Fi",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Drama",
          "Sci-Fi",
        ],
        "className": "movie-genre",
      },
      "ref": null,
      "rendered": Array [
        "Drama",
        "Sci-Fi",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
